extensions/machine-learning-services/src/common/queryRunner.ts

```diff
@@ -86,7 +86,21 @@ export class QueryRunner {
 	 * Returns true if python installed in the give SQL server instance
 	 */
 	public async isPythonInstalled(connection: azdata.connection.ConnectionProfile): Promise<boolean> {
-		let result = await this.runQuery(connection, checkPythonInstalledQuery);
+		return this.isLanguageInstalled(connection, 'Python');
+	}
+
+	/**
+	 * Returns true if R installed in the give SQL server instance
+	 */
+	public async isRInstalled(connection: azdata.connection.ConnectionProfile): Promise<boolean> {
+		return this.isLanguageInstalled(connection, 'R');
+	}
+
+	/**
+	 * Returns true if language installed in the give SQL server instance
+	 */
+	private async isLanguageInstalled(connection: azdata.connection.ConnectionProfile, language: string): Promise<boolean> {
+		let result = await this.runQuery(connection, checkLanguageInstalledQuery.replace('#LANGUAGE#', language));
```

Comment:
As with the config - why are you doing the manual replace like this instead of using string formatting? That's a much more standard way of handling stuff like this

Reply:
you mean string format like `string value {param}` ? I have the scripts as template. that's why I'm using it like this, if there's a better way can you send me a sample please? 

---
In reply to: [362901144](https://github.com/microsoft/azuredatastudio/pull/8796#discussion_r362901144) [](ancestors = 362901144)

Reply:
Sorry, forgot this was in an extension (VS Code provides a format function for doing this in the core code). This is fine then (although it would be awesome if Javascript actually provided this natively...)

Depending on usage you can also use string interpolation instead - which I personally prefer as long as the string is only used in one place.

Reply:
I'm moving all these scripts to sqltools anyway. if you're ok I keep it as it and in next PR I use sqltoolsservice 

---
In reply to: [363421524](https://github.com/microsoft/azuredatastudio/pull/8796#discussion_r363421524) [](ancestors = 363421524)